src/lib/coerce.js

```diff
@@ -39,7 +39,7 @@ exports.valObjects = {
         requiredOpts: ['values'],
         otherOpts: ['dflt', 'coerceNumber', 'arrayOk'],
         coerceFunction: function(v, propOut, dflt, opts) {
-            if(opts.coerceNumber) v = +v;
+            if(opts.coerceNumber && isNumeric(v)) v = +v;
```

Comment:
:tiger2: we should add a test for this in this [block](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/lib_test.js#L438).